cli/npm/resolution.rs

```diff
@@ -105,13 +105,14 @@ impl std::fmt::Display for NpmPackageReference {
   }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[derive(Debug, Clone, PartialOrd, Ord, PartialEq, Eq, Hash)]
 pub struct NpmPackageId {
   pub name: String,
   pub version: semver::Version,
 }
 
```

Comment:
I guess it was landed in an earlier patch but I'm unclear on why `NpmVersionMatcher` is a trait instead of a struct? It seems there's only one implementation?

Reply:
There's two implementations. One is the for dependencies specified in package.json and the other is for the `npm:<package>@<version-req>`. The one in npm version requirement is more restrictive at the moment and only uses the `semver` crate. We could open it up in the future to supporting stuff like `npm:<package>@11.0.1 || > 15`, but for now I made it not allowed there (I'm not sure what version requirement we want to eventually use in npm specifiers).

Reply:
Makes sense! This would be great to have as a code comment on `NpmVersionMatcher`